### ISSUE TITLE
Install without cocoapods podfile

### ISSUE THREAD

**Issue by pstanton** [react-native-community/react-native-maps#2413]
### Is this a bug report?

Yes

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?

Yes

### Environment

		"base64-js": "^1.3.0",
		"buffer": "^5.1.0",
		"jspack": "^0.0.4",
		"mobx": "^4.2.1",
		"mobx-logger": "^0.7.0",
		"mobx-react": "^5.1.2",
		"prop-types": "^15.6.1",
		"react": "^16.3.1",
		"react-native": "^0.55.4",
		"react-native-ble-plx": "^0.9.0",
		"react-native-camera": "^1.1.4",
		"react-native-communications": "^2.2.1",
		"react-native-device-info": "^0.21.5",
		"react-native-elements": "^0.19.1",
		"react-native-exception-handler": "^2.8.9",
		"react-native-image-resizer": "^1.0.0",
		"react-native-maps": "^0.21.0",
		"react-native-vector-icons": "^4.6.0"

### Steps to Reproduce

I've had a working installation on android using all of the listed dependencies, and on ios, everything but react-native-maps .. working for some time. Other dependencies were all added using their "manual installation" method - because I don't trust linking generally.

I've never before used cocoapods and in order to follow your installation instructions, had to install it. I've followed your installation instructions to a T and hoped to get my project building with maps finally on ios.

However now I only get "build failed" and a whole bunch of errors regarding my other dependencies. Some I've managed to work through by running clean, yarn and re-installation (manually) again. However the latest issue I cannot seem to fix. It is the same as [this](https://github.com/react-native-community/react-native-camera/issues/1407).

The reason I'm posting here, is because this was not an issue before going through your installation instructions. The spaghetti of npm+yarn+xcode is pretty damn hard to deal with at the best of times, adding another layer of spaghetti (cocoapods) seems to be the nail in the coffin.

_Is there a way to install your project without using cocoapods, and if so please document it. I've tried instructions posted by users in other issue feeds with no luck._

**Comment by pstanton** [comment]
I've managed to solve all my build issues by converting `react-native-device-info` and `react-native-camera` over to cocoapods installs.

**Comment by jamesone** [comment]
@pstanton There are open PRs which will provide the use of `react-native link`

- https://github.com/react-community/react-native-maps/pull/2310
- https://github.com/react-community/react-native-maps/pull/2396

**Comment by pstanton** [comment]
@thanhdevapp it took me hours of trial and error. this architecture of multiple dependency management and build systems all conflicting is the worst. react-native-ios is worse than react-native-android but android isn't fantastic either. sorry, can't help.

**Comment by thanhdevapp** [comment]
@pstanton thanks you, so now, can't install on ios right?

**Comment by JanErikFoss** [comment]
You can use version ^0.20.1 in the meantime, it works without cocoapods.

That's what I do until they release a new version.

**Comment by ProteanDev** [comment]
@JanErikFoss how do I use GoogleMaps instead of AppleMaps for iOS in 0.20.1?

**Comment by ProteanDev** [comment]
@JanErikFoss ugh... How do we use Google Maps for iOS with ^0.20.1. Docs are a bit fussy about that.

**Comment by JanErikFoss** [comment]
@ProteanDev I don't remember much about this issue, but I don't think you'll be able to use Google Maps with 0.20.1 and without cocoapods. From what I can recall, Google Maps support was the big issue that made cocoapods required.

**Comment by ProteanDev** [comment]
@JanErikFoss even if I already successfully installed GoogleMaps framework manually?

**Comment by JanErikFoss** [comment]
> @JanErikFoss even if I already successfully installed GoogleMaps framework manually?

I really don't know enough to give you an answer.

**Comment by MarshMelon** [comment]
Did anyone find a solution?

**Comment by PurnimaNaik** [comment]
@niranjanbprajapati did you find a solution to the `Use of undeclared identifier 'AIRGoogleMapOverlay'` error? I am facing the same issue

**Comment by MarshMelon** [comment]
I add to my podfile this pods install this
   # react-native-maps dependencies
    pod 'react-native-maps', :path => '../node_modules/react-native-maps'
    pod 'react-native-google-maps', path: rn_maps_path  # Unomment this line if you want to support GoogleMaps on iOS
    pod 'GoogleMaps'  # Uncomment this line if you want to support GoogleMaps on iOS
    pod 'Google-Maps-iOS-Utils' # Uncomment this line if you want to support GoogleMaps on iOS

  end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        if target.name == 'react-native-google-maps'
            target.build_configurations.each do |config|
                config.build_settings['CLANG_ENABLE_MODULES'] = 'No'
            end
        end
        if target.name == "React"
            target.remove_from_project
        end
    end
end

import @import GoogleMaps; and add  
 [GMSServices provideAPIKey:@"Your api key"]; to AppDelegate.m

and run

**Comment by PurnimaNaik** [comment]
refer #2573 for the solution to `Use of undeclared identifier 'AIRGoogleMapOverlay'` error

**Comment by Vic-L** [comment]
I got it working on iOS without Podfile, and I think I will document it here.

tl;dr; There is 1 key step missing, which is to add `Framework Search Path` to `AirMaps.xcodeproj`.

The installation guide for both `Google` and `react-native-maps` are not very clear. I did quite a number of trial and errors and here is my version.

1. Install [`Google Maps iOS SDK`](https://developers.google.com/maps/documentation/ios-sdk/start) first. NOTE: whenever you copy your files to Xcode, use `Copy items if needed` and `Create groups`. The instruction in the manual is referencing an old version. NOTE AGAIN: I copied the `GoogleMap.bundle` from the `GoogleMaps.framework` in my folder (after first copying the 3 frameworks), not from the `GoogleMaps.framework` in my `Downloads`. NOTE ONE MORE TIME: Google steps mention to use `@import GoogleMaps;` but the `react-native-maps` mention to use `#import <GoogleMaps/GoogleMaps.h>` in `AppDelegate.m`. Both seem to be working, but it might be a caching issue on my side. Do trial and error.
2. Remember to add the snippet below to your `Info.plist`; this is/was under step 6 of the google Maps iOS SDK installation manual.
```
<key>LSApplicationQueriesSchemes</key>
<array>
    <string>googlechromes</string>
    <string>comgooglemaps</string>
</array>
```
3. The `react-native-maps` mentioned a `postinstall` script. That will solve the bug of `AirGoogleMaps dir must be added to your xCode project to support GoogleMaps on IOS`. For me, i changed `REPLACE_ME_RELATIVE_PATH_TO_GOOGLE_MAPS_INSTALL` to `ios/GoogleMaps/**` with the folder structure as shown below:
<img width="269" alt="appdelegate_m" src="https://user-images.githubusercontent.com/6149863/52914386-e5d2d580-330a-11e9-99a2-fe5614f50217.png">
4. Add the `Framework Search Path` under `AirMaps.xcodeproj`'s `Target` -> `Build Settings` -> `Search Paths`. Base on my folder structure, i added `$(SRCROOT)/../../../../ios/GoogleMaps` and made it `recursive`. This solved those `<GoogleMaps/GoogleMaps.h> file not found` errors.

It will be great if I can get some feedback if this works on your machines. I can make a PR to update the docs and maybe the postinstall script to add the framework.

**Comment by alvelig** [comment]
But I'm guessing pod will be the only supported way to install things. What are the problems faced while using it? Could you, guys, comment?

**Comment by Vic-L** [comment]
I had some issues with Cocoapods with other libraries when I first started out with RN. I can't really exactly remember but it's along the same line as the [OP's comment](https://github.com/react-native-community/react-native-maps/issues/2413#issuecomment-411559624).

**Comment by alvelig** [comment]
@Vic-L most of those I've seen were to start without Podfile, link libraries, then add Pods, install and get all sort of problems. :D  If you move to Podfile, you have to move everything to it, otherwise `react-native link` won't work as expected.

**Comment by Vic-L** [comment]
I will like to condemn the way I did to get the library working without cocoapods. It is actually changing the files in the `AirMap` libraries in the `node_modules`. This will cause dependency problems from git clones and stuff, which I am facing right now trying to use build with BitRise. :(

I guess i will need to use cocoapods.

**Comment by Vic-L** [comment]
I found a workaround which does not require fiddling with the 
`Framework Search Path` that will change the files inside `node_modules` which is not ideal when working in a team.

After running `react-native link react-native-maps`, check where the `Framework Search Path` is pointing to in `AirMaps.xcodeproj` in your project directory. Place all the google maps files into that directory without any further sub directories. Check that physically the files are in the same location as reflected in the file structure shown in Xcode side bar.

tl;dr; the gist is to choose the correct place to place the google map files based on where `AirMaps.xcodeproj` is looking at by default

**Comment by VilleMiekkoja** [comment]
> I found a workaround which does not require fiddling with the
> `Framework Search Path` that will change the files inside `node_modules` which is not ideal when working in a team.
> 
> After running `react-native link react-native-maps`, check where the `Framework Search Path` is pointing to in `AirMaps.xcodeproj` in your project directory. Place all the google maps files into that directory without any further sub directories. Check that physically the files are in the same location as reflected in the file structure shown in Xcode side bar.
> 
> tl;dr; the gist is to choose the correct place to place the google map files based on where `AirMaps.xcodeproj` is looking at by default. This should result in a messy arrangement of files.

But where do you find the AirMaps.xcodeproj file/folder? Somewhere in node_modules?
